File: client/common/model/door/door.ts

export class ActionCommand {
    key : string;
    code: string;
    version: string;

    constructor(key: string, code: string, version: string) {
        this.key = key;
        this.code = code;
        this.version = version;
    }
}


export class DoorFileRecord {
    id: number|undefined;
    source: string|undefined;
    fileId: string|undefined;
    resourceId: number|undefined;
    fileType: string|undefined;
    fileName: string|undefined;
    fileUrl: string|undefined;
    fileSize: number|undefined;
    folderId: string|undefined;
    fileKey: string|undefined;
    pix: string|undefined;

    constructor(id?: number, source?: string, fileId?: string, resourceId?: number, fileType?: string, fileName?: string, fileUrl?: string, fileSize?: number, folderId?: string, fileKey?: string, pix?: string) {
        this.id = id;
        this.source = source;
        this.fileId = fileId;
        this.resourceId = resourceId;
        this.fileType = fileType;
        this.fileName = fileName;
        this.fileUrl = fileUrl;
        this.fileSize = fileSize;
        this.folderId = folderId;
        this.fileKey = fileKey;
        this.pix = pix;
    }
}

export class DoorRecord {
    id: number | undefined;
    doorKey: string;
    url: string;
    itemKey: string;
    type: string;
    data: string;

    constructor(id: number | undefined, doorKey: string, url: string, itemKey: string, type: string, data: string) {
        this.id = id;
        this.doorKey = doorKey;
        this.url = url;
        this.itemKey = itemKey;
        this.type = type;
        this.data = data;
    }
}

export class SearchSkuRecord {
    id: number | undefined;
    type: string;
    title: string;
    skuId: string | undefined;

    constructor(id: number | undefined, type: string, title: string, skuId?: string) {
        this.id = id;
        this.type = type;
        this.title = title;
        this.skuId = skuId;
    }
}

export class DoorSkuCatProp {
    id: number | undefined;
    source: string;
    itemKey: string;
    propKey: string;
    propValue: any;

    constructor(id: number | undefined, source: string, itemKey: string, propKey: string, propValue: any) {
        this.id = id;
        this.source = source;
        this.itemKey = itemKey;
        this.propKey = propKey;
        this.propValue = propValue;
    }
}

File: server/internal/door/models/door.go
package models

import (
	"server/common/base"
	"server/common/middleware/database"
)

type DoorRecord struct {
	database.BaseEntity
	ItemKey    string    `json:"item_key"`
	Type       string    `json:"type"`
	DoorKey    string    `json:"door_key"`
	Url        string    `json:"url"`
	OssUrl     string    `json:"oss_url"`
	Version    string    `json:"version"`
	ExpireTime base.Time `json:"expire_time"`
}

func (u *DoorRecord) TableName() string {
	return "door_record"
}

type DoorCatProp struct {
	database.BaseEntity
	ItemKey   string `json:"item_key"`
	Source    string `json:"source"`
	PropKey   string `json:"prop_key"`
	PropValue string `json:"prop_value"`
}

func (u *DoorCatProp) TableName() string {
	return "door_cat_prop"
}

type DoorFileRecord struct {
	database.BaseEntity
	ResourceId uint64 `json:"resource_id"`
	Source     string `json:"source"`
	FileType   string `json:"file_type"`
	FileSize   uint64 `json:"file_size"`
	FolderId   string `json:"folder_id"`
	FileId     string `json:"file_id"`
	FileUrl    string `json:"file_url"`
	FileName   string `json:"file_name"`
	FileKey    string `json:"file_key"`
	Pix        string `json:"pix"`
}

type SearchSkuRecord struct {
	database.BaseEntity
	SkuId    string `json:"sku_id"`
	Type     string `json:"type"`
	Title    string `json:"title"`
	PddSkuId string `json:"pdd_sku_id"`
}

func (f *DoorFileRecord) TableName() string {
	return "door_file_record"
}

func (s *SearchSkuRecord) TableName() string {
	return "search_sku_record"
}

type DoorCategory struct {
	database.BaseEntity
	PddCatId  string `json:"pdd_cat_id"`
	TbCatId   string `json:"tb_cat_id"`
	TbCatName string `json:"tb_cat_name"`
}

func (u *DoorCategory) TableName() string {
	return "door_category"
}
